### PR TITLE
Use dispatcher methods for fetch and fetchById in tests

### PR DESCRIPTION
**Pull request by Cypher1**

### CODE REVIEW

File: javatests/arcs/core/entity/HandleManagerTestBase.kt [hunk 1]
```diff
@@ -549,22 +549,22 @@ open class HandleManagerTestBase {
             assertThat(handle.size()).isEqualTo(2)
             assertThat(handle.isEmpty()).isEqualTo(false)
             assertThat(handle.fetchAll()).containsExactly(entity1, entity2)
-            assertThat(handle.fetchById(entity1.entityId)).isEqualTo(entity1)
-            assertThat(handle.fetchById(entity2.entityId)).isEqualTo(entity2)
+            assertThat(handle.dispatchFetchById(entity1.entityId)).isEqualTo(entity1)
+            assertThat(handle.dispatchFetchById(entity2.entityId)).isEqualTo(entity2)
```
Comment (mykmartin) [inline javatests/arcs/core/entity/HandleManagerTestBase.kt:553]: Actually not needed for this test; this block is already running on the dispatcher (see line 535).

File: javatests/arcs/core/host/AbstractArcHostTest.kt [hunk 2]
```diff
@@ -177,7 +178,7 @@ open class AbstractArcHostTest {
         host.getFooHandle().dispatchStore(entity)
         var storedEntity = EntityBase("EntityBase", DummyEntity.SCHEMA)
         storedEntity.setSingletonValue("text", "Watson")
-        assertThat((host.getFooHandle() as ReadWriteSingletonHandle<EntityBase>).fetch()?.equals(
+        assertThat((host.getFooHandle() as ReadWriteSingletonHandle<EntityBase>).dispatchFetch()?.equals(
```
Comment (mykmartin) [inline javatests/arcs/core/host/AbstractArcHostTest.kt:181]: While you're touching this, can you fix a bug? The assert is bracketed incorrectly so it's not actually testing anything...

```
assertThat((host.getFooHandle() as ReadWriteSingletonHandle<EntityBase>).dispatchFetch()).equals(storedEntity)
```

**Review by mykmartin — APPROVED** [review]

File: javatests/arcs/core/entity/HandleManagerTestBase.kt (hunk 1 above)
Comment (Cypher1) [reply javatests/arcs/core/entity/HandleManagerTestBase.kt:553]: Hmm, I'd agree but the test failed without them on my machine.
Any ideas as to why?

File: javatests/arcs/core/host/AbstractArcHostTest.kt (hunk 2 above)
Comment (Cypher1) [reply javatests/arcs/core/host/AbstractArcHostTest.kt:181]: No problem

File: javatests/arcs/core/entity/HandleManagerTestBase.kt (hunk 1 above)
Comment (mykmartin) [reply javatests/arcs/core/entity/HandleManagerTestBase.kt]: I would be very surprised and quite worried if calling `fetchById` inside a dispatcher context block caused problems, given that this is all the dispatch version of the function does...

Comment (Cypher1) [reply javatests/arcs/core/entity/HandleManagerTestBase.kt]: Yeah, can confirm that these ones were fine. I was a little overzealous with the `find and replace`.

**Comment by Cypher1** [comment]
Failed a performance test (to me that sounds flakey, re-running to see)
/gcbrun

**Pull request merged**